numba/unicode.py

```diff
@@ -1267,6 +1267,52 @@ def impl(a):
         return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L10765-L10774    # noqa: E501
+@overload_method(types.UnicodeType, 'capitalize')
+def unicode_capitalize(data):
+    """Implements str.capitalize()"""
+    def impl(data):
+        # https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L9737-L9759    # noqa: E501
```

Comment:
Impl seems good, though lacks an ASCII encoding fast-path, so performance will likely be poor in this case.

Reply:
Added ascii capitalization.
@stuartarchibald could you please take the next round of review?